src/ring_theory/ideal/operations.lean

```diff
@@ -515,34 +519,43 @@ theorem is_prime.inf_le {I J P : ideal R} (hp : is_prime P) :
 ⟨λ h, hp.mul_le.1 $ le_trans mul_le_inf h,
 λ h, or.cases_on h (le_trans inf_le_left) (le_trans inf_le_right)⟩
 
-theorem is_prime.prod_le {s : finset ι} {f : ι → ideal R} {P : ideal R}
-  (hp : is_prime P) (hne: s.nonempty) :
-  s.prod f ≤ P ↔ ∃ i ∈ s, f i ≤ P :=
-suffices s.prod f ≤ P → ∃ i ∈ s, f i ≤ P,
-  from ⟨this, λ ⟨i, his, hip⟩, le_trans prod_le_inf $ le_trans (finset.inf_le his) hip⟩,
+theorem is_prime.multiset_prod_le {s : multiset (ideal R)} {P : ideal R}
+  (hp : is_prime P) (hne : s ≠ 0) :
+  s.prod ≤ P ↔ ∃ I ∈ s, I ≤ P :=
+suffices s.prod ≤ P → ∃ I ∈ s, I ≤ P,
+  from ⟨this, λ ⟨i, his, hip⟩, le_trans multiset_prod_le_inf $
+    le_trans (multiset.inf_le his) hip⟩,
 begin
   classical,
-  obtain ⟨b, hb⟩ : ∃ b, b ∈ s := hne.bex,
-  obtain ⟨t, hbt, rfl⟩ : ∃ t, b ∉ t ∧ s = insert b t,
-  from ⟨s.erase b, s.not_mem_erase b, (finset.insert_erase hb).symm⟩,
-  revert hbt,
+  obtain ⟨b, hb⟩ : ∃ b, b ∈ s := multiset.exists_mem_of_ne_zero hne,
+  obtain ⟨t, rfl⟩ : ∃ t, s = b ::ₘ t,
+  from ⟨s.erase b, (multiset.cons_erase hb).symm⟩,
   refine t.induction_on _ _,
```

Comment:
`indunction t using finset.induction_on` would probably save you needing to `intros` below, but it doesn't matter.

Reply:
Not precisely, since we still need to `intros h` in the inductive step. So I'll keep it as-is.